pkg/scheduler/algorithm/priorities/metadata.go

```diff
@@ -123,32 +123,52 @@ func getSelector(pod *v1.Pod, sl corelisters.ServiceLister, cl corelisters.Repli
 		}
 	}
 
-	if rcs, err := cl.GetPodControllers(pod); err == nil {
-		for _, rc := range rcs {
-			labelSet = labels.Merge(labelSet, rc.Spec.Selector)
+	rcPod, rsPod, stsPod := false, false, false
+	for _, ref := range pod.OwnerReferences {
+		switch ref.Kind {
+		case "ReplicationController":
+			rcPod = true
+		case "ReplicaSet":
+			rsPod = true
+		case "StatefulSet":
+			stsPod = true
```

Comment:
kind is not sufficient to partition owner reference, you need apiVersion and kind to match

Reply:
addressed, thanks.